src/raiden_libs/register_service.py

```diff
@@ -0,0 +1,142 @@
+    CONTRACT_CUSTOM_TOKEN,
+    CONTRACT_SERVICE_REGISTRY,
+    CONTRACT_USER_DEPOSIT,
+)
+from raiden_libs.cli import blockchain_options, common_options
+from raiden_libs.contract_info import CONTRACT_MANAGER
+from raiden_libs.utils import private_key_to_address
+
+log = structlog.get_logger(__name__)
+
+
+@blockchain_options(contracts=[CONTRACT_SERVICE_REGISTRY, CONTRACT_USER_DEPOSIT])
+@click.command()
+@click.option(
+    "--deposit",
+    default=100 * 10 ** 18,  # 100 RDN
+    type=click.IntRange(min=1),
+    help="Amount of tokens to deposit in the ServiceRegistry",
+)
+@click.option("--service-url", type=str, help="URL for the services to register")
+@common_options("register_service")
+def main(
+    private_key: str,
+    state_db: str,
+    web3: Web3,
+    contracts: Dict[str, Contract],
+    start_block: BlockNumber,
+    deposit: TokenAmount,
+    service_url: str,
+) -> None:
+    # Add middleware to sign transactions by default
+    web3.middleware_stack.add(construct_sign_and_send_raw_middleware(private_key))
+
+    service_address = private_key_to_address(private_key)
+    log.info("Running service registration script", account_address=service_address)
+
+    service_registry_contract = contracts[CONTRACT_SERVICE_REGISTRY]
+
+    # check if already registered
+    current_deposit = service_registry_contract.functions.deposits(service_address).call()
+    current_url = service_registry_contract.functions.urls(service_address).call()
+    log.info(
+        "Current ServiceRegistry information for service address",
+        service_address=service_address,
+        current_deposit=current_deposit,
+        current_url=current_url,
+    )
+
+    # Register if not yet done
+    if current_deposit <= 0:
+        log.info("Address not registered in ServiceRegistry")
+
+        custom_token_address = contracts[CONTRACT_USER_DEPOSIT].functions.token().call()
+        custom_token_contract = web3.eth.contract(
+            address=custom_token_address,
+            abi=CONTRACT_MANAGER.get_contract_abi(CONTRACT_CUSTOM_TOKEN),
+        )
+
+        # Check current token balance
+        account_balance = custom_token_contract.functions.balanceOf(service_address).call()
+        log.info("Current account balance", balance=account_balance, desired_deposit=deposit)
+
+        # mint token if necessary
+        if account_balance < deposit:
+            run_task(
+                web3=web3,
+                service_address=service_address,
+                function_call=custom_token_contract.functions.mint(deposit),
+                task_name="Minting new Test RDN tokens",
+            )
+
+            account_balance = custom_token_contract.functions.balanceOf(service_address).call()
+            log.info("Updated account balance", balance=account_balance, desired_deposit=deposit)
+
+        current_allowance = custom_token_contract.functions.allowance(
+            service_address, service_registry_contract.address
+        ).call()
```

Comment:
Do we need the allowance check? Usually there should be now allowance left, and if there is, setting it again won't cause more problems than the wasted gas.

Reply:
You mean to get rid of the check and call `allow` unconditionally?

Reply:
`approve` unconditionally, yes. From my understanding, that's what will happen in > 99% of the cases, anyway.